controllers/rekor/actions/server/createTree.go

```
@@ -38,12 +39,20 @@ func (i createTrillianTreeAction) Handle(ctx context.Context, instance *rhtasv1a
 		return i.StatusUpdate(ctx, instance)
 	}
 	var err error
+	var tree *trillian.Tree
+	var trillUrl string
 
-	trillUrl, err := k8sutils.GetInternalUrl(ctx, i.Client, instance.Namespace, trillian.LogserverDeploymentName)
-	if err != nil {
-		return i.Failed(err)
+	switch {
+	case instance.Spec.Trillian.Port == nil:
+		err = fmt.Errorf("%s: %w", i.Name(), utils.TrillianPortNotSpecified)
```

You should throw the error. It could be swallowed by `tree, err = common.CreateTrillianTree(ctx, "rekor-tree", trillUrl, constants.CreateTreeDeadline)`.

---

this check is required for building of `trillUrl`. It will fails on fatal error before it will be passed to CreateTrillianTree function if it is nil. The reason is that `Port` has to be dereferenced and uset to crreate `trillUrl`